server/lib/routes/post-csp.js

```diff
@@ -8,7 +8,42 @@
 
 'use strict';
 const _ = require('lodash');
+const joi = require('joi');
 const url = require('url');
+const validation = require('../validation');
+
+const INTEGER_TYPE = validation.TYPES.INTEGER;
+const STRING_TYPE = validation.TYPES.STRING;
+const URL_TYPE = validation.TYPES.URL;
+
+const BODY_SCHEMA = {
+  'csp-report': joi.object().keys({
+    // CSP 2, 3 required
+    'blocked-uri': URL_TYPE.allow('self').required(),
```

Comment:
I tried this in Chrome and via an unsafe eval like:
```
$('body').html('<script>alert(1)</script>')
```
csp failed to submit:


```
fxa-content-server.server.main.ERROR: {"error":"\"blocked-uri\" must be a valid uri with a scheme matching the http|https pattern","op":"validation.error","path":"/_/csp-violation"}
```

Payload: 

```
{"csp-report":{"document-uri":"http://127.0.0.1:3030/confirm","referrer":"","violated-directive":"script-src","effective-directive":"script-src","original-policy":"connect-src 'self' http://127.0.0.1:9000 http://127.0.0.1:9010 http://127.0.0.1:1111 http://127.0.0.1:1114; default-src 'self'; font-src 'self'; img-src 'self' data: https://secure.gravatar.com http://127.0.0.1:1112; media-src blob:; object-src 'none'; report-uri /_/csp-violation; script-src 'self'; style-src 'self' 'sha256-9n6ek6ecEYlqel7uDyKLy6fdGNo3vw/uScXSq9ooQlk='","disposition":"enforce","blocked-uri":"inline","line-number":83,"column-number":12,"source-file":"http://127.0.0.1:3030/bower_components/jquery/dist/jquery.js","status-code":200}}
```